src/test/java/fi/otavanopisto/kuntaapi/test/AbstractPtvMocker.java

```diff
@@ -0,0 +1,493 @@
+      if (resource.getStatus().equals(status)) {
+        result.add(resource);
+      }
+    }
+    
+    return result;
+  }
+  
+  public GuidPage getGuidPage(MockedResourceStatus status) {
+    GuidPage result = new GuidPage();
+    
+    for (MockedResource<String, R> mockedResource : getMockedResources(status)) {
+      VmOpenApiItem item = new VmOpenApiItem();
+      item.setId(mockedResource.getId());
+      item.setName(mockedResource.getId());
+      result.addItem(item);
+    }
+    
+    result.setPageCount(1);
+    result.setPageNumber(0);
+    result.setPageSize(1000);
+    
+    return result;
+  }
+  
+  public boolean isMocked(String id) {
+    return resources.containsKey(id);
+  }  
+  
+  public void setStatus(String id, MockedResourceStatus status) {
+    MockedResource<String, R> resource = resources.get(id);
+    if (resource.getStatus() == status) {
+      return;
+    }
+    
+    if (started && resource.getCurrentMappings() != null) {
+      for (MappingBuilder mapping : resource.getCurrentMappings()) {
+        removeStub(mapping);
+      }
+    }
+        
+    resource.setStatus(status);
+    
+    if (started) {
+      for (MappingBuilder mapping : resource.getCurrentMappings()) {
+        stubFor(mapping);
+      }
+      
+      updateStatusLists();
+      updateSubMockerStatuses(id, status);
+    }
+  }
+  
+  @SuppressWarnings ("squid:S1301")
+  public void mockAlternative(String id, R alternative) {
+    List<MappingBuilder> alternativeMappings = new ArrayList<>();
+    MockedResource<String, R> resource = resources.get(id);
+    
+    List<MappingBuilder> okMappings = resource.getMappings(MockedResourceStatus.OK);
+    for (MappingBuilder okMapping : okMappings) {
+
+      if (started) {
+        removeStub(okMapping);
+      }
+      
+      MappingBuilder alternativeMapping = null;
+      String method = okMapping.build().getRequest().getMethod().getName();
+      
+      switch (method) {
+        case "GET":
+          alternativeMapping = okMapping.willReturn(createOkGetReturn(alternative));
+        break;
+        case "HEAD":
+          alternativeMapping = okMapping.willReturn(createOkHeadReturn(alternative));
+        break;
+        default:
+          fail(String.format("Unknown method %s", method));
+        break;
+      }
+      
+      if (started) {
+        stubFor(alternativeMapping);
+      } 
+      
+      alternativeMappings.add(alternativeMapping);
+    }
+    
+    resource.setResource(alternative);
+    resource.updateStatusMappings(MockedResourceStatus.OK, alternativeMappings);
+  }
+
+  public AbstractPtvMocker<R> mock(String... ids) {
+    for (String id : ids) {
+      try {
+        if (!isMocked(id)) {
+          mockEntity(id, readEntity(id));
+        } else {
+          setStatus(id, MockedResourceStatus.OK);
+        }
+      } catch (JsonProcessingException e) {
+        System.err.println(String.format("Failed to read %s of %s", id, getName()));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)